src/test/librbd/test_librbd.cc

```diff
@@ -3533,12 +3533,22 @@ TEST_F(TestLibRBD, TestClone2)
                                            &snap_namespace_type));
   ASSERT_EQ(RBD_SNAP_NAMESPACE_TYPE_TRASH, snap_namespace_type);
 
+  rbd_snap_group_namespace_t group_snap_type;
+  ASSERT_EQ(0, rbd_snap_get_group_namespace(parent, snaps[0].id,
+                                            &group_snap_type,
+                                            sizeof(group_snap_type)));
```

Comment:
You may call `rbd_snap_get_group_namespace` for a group snapshot, while you are calling it for a trash snapshot. It will fail with `-EINVAL` as you may see in the jenkins `make check` log.

Reply:
nice, it's helpful to a ceph noob, thank you.

Reply:
So I think you may add `rbd_snap_get_group_namespace` test to `test_Groups.cc`,  "TestGroup.add_snapshot" and its c++ analog to "TestGroup.add_snapshotPP".